src/test/java/pl/zaradny/springApp/api/ProductEndpointTest.java

```diff
@@ -62,6 +65,43 @@ public void shouldGetListOfAllProducts(){
         assertTrue(result.getBody().getProducts().contains(prd2));
     }
 
+    @Test
+    public void shouldGetProductsWithTagFromRequestParam(){
+        //given
+        List<TagDto> tagsList = new ArrayList<>();
+        tagsList.add(new TagDto("tag"));
+        ProductResponseDto prd1 = productFacade.create(new ProductRequestDtoBuilder("product",
+                new PriceDto("100", "PLN")).withTags(tagsList).build());
+        ProductResponseDto prd2 = productFacade.create(new ProductRequestDtoBuilder("product",
+                new PriceDto("150", "EUR")).build());
+        final String url = productsUrl + "?tag=tag";
+
```

Comment:
Nie żałuj linii ;) widać dość często że upakowujesz w jednym ciągu kolejne wywołania builderów lub argumenty konstruktora, a naprawdę warto przenosić te elementy do nowych linii, to bardzo mocno wpływa na czytelność (chociaż złośliwi twierdzą, że nie powinno być już ograniczeń długości linii kiedy mamy takie wypasione monitory panoramiczne) :D
```
ProductResponseDto prd1 = productFacade.create(
    new ProductRequestDtoBuilder("product", new PriceDto("100", "PLN"))
        .withTags(tagsList)
        .build()
);
```

natomiast tworzenie listy tagów możesz bez problemów z inlinowować ;) 
`Lists.asList(new TagDto("tag"))` i nawet włożyć odrazu w withTags, skoro nie jest już nigdzie więcej używane ;)